sdk/storage/storage-blob/src/StorageClient.ts

```diff
@@ -89,10 +89,11 @@ export abstract class StorageClient {
     for (const factory of this.pipeline.factories) {
       if (
         (isNode && factory instanceof StorageSharedKeyCredential) ||
-        factory instanceof AnonymousCredential ||
-        isTokenCredential(factory)
+        factory instanceof AnonymousCredential
       ) {
         this.credential = factory;
+      } else if (isTokenCredential((factory as any).credential)) {
```

Comment:
>(isTokenCredential((factory as any).credential)) { [](start = 16, length = 50)

To parser TokenCredential from a pipeline factory arrary. It only works if the TokenCredentialPolicy has attached "credential" property. --> Only if the pipeline objected is created from `newPipeline()` method which will do the attachment.

Please add a comment about that.

Reply:
Done.